test/generators/go/GoGenerator.spec.ts

```diff
@@ -5,6 +5,28 @@ describe('GoGenerator', () => {
   beforeEach(() => {
     generator = new GoGenerator();
   });
+
+  test('should not render reserved keyword', async () => {
+    const doc = {
+      $id: 'Address',
+      type: 'object',
+      properties: {
+        enum: { type: 'string' },
```

Comment:
I'm a bit confused. Even though `enum` is not a real keyword in go, you added it to the list here: https://github.com/asyncapi/modelina/pull/653/files#r813939730 so I would expect this test to fail since `Enum` would never be possible as field name. 

Anyway, as mentioned earlier, `enum` is not a keyword. What do you think about generating a test that takes all the keywords defined at `RESERVED_GO_KEYWORDS` constant testing them one by one?

Reply:
Do you mean testing all 25 reserved keywords in a single test or creating a test for each reserved keyword?

Reply:
I think one single test that tests all of them at once iterating them should do the job here. WDYT?

Reply:
I agree, I have added the single iteration for all please review and let me know If I implemented it correctly or not 🙂